x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityLegacyCrossClusterApiKeysWithDlsFlsIT.java

```diff
@@ -0,0 +1,269 @@
+                { "index": { "_index": "shared-metrics" } }
+                { "name": "metric4" }
+                """));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+        }
+
+        // Query cluster -- test searching works (the API key is valid)
+        {
+            final var searchRequest = new Request(
+                "GET",
+                String.format(
+                    Locale.ROOT,
+                    "/%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    randomFrom("my_remote_cluster", "*", "my_remote_*"),
+                    randomFrom("shared-metrics", "*"),
+                    randomBoolean()
+                )
+            );
+            final Response response = performRequestWithRemoteMetricUser(searchRequest);
+            assertOK(response);
+            final SearchResponse searchResponse = SearchResponseUtils.parseSearchResponse(responseAsParser(response));
+            try {
+                final List<String> actualIndices = Arrays.stream(searchResponse.getHits().getHits())
+                    .map(SearchHit::getIndex)
+                    .collect(Collectors.toList());
+                assertThat(Set.copyOf(actualIndices), containsInAnyOrder("shared-metrics"));
+            } finally {
+                searchResponse.decRef();
+            }
+        }
+
+        // make API key invalid
+        addDlsQueryToApiKeyDoc(crossClusterAccessApiKeyId);
+
+        // since we updated the API key doc directly, caches need to be clearer manually -- this would also happen during a rolling restart
+        // to the FC, during an upgrade
+        assertOK(performRequestAgainstFulfillingCluster(new Request("POST", "/_security/role/*/_clear_cache")));
+        assertOK(performRequestAgainstFulfillingCluster(new Request("POST", "/_security/api_key/*/_clear_cache")));
+
+        // check that GET still works
+        getCrossClusterApiKeys(crossClusterAccessApiKeyId);
+        {
+            final var searchRequest = new Request(
+                "GET",
+                String.format(
+                    Locale.ROOT,
+                    "/%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    "my_remote_cluster",
+                    "shared-metrics",
+                    randomBoolean()
+                )
+            );
+            // TODO test skip_unavailable set and not set
+            final ResponseException ex = expectThrows(ResponseException.class, () -> performRequestWithRemoteMetricUser(searchRequest));
+            assertThat(
+                ex.getMessage(),
+                containsString("search does not support document or field level security if replication is assigned")
+            );
+            assertThat(ex.getResponse().getStatusLine().getStatusCode(), equalTo(400));
+        }
+    }
+
+    // TODO a test for CCR follow (and ideally auto-follow) -- however, this is less critical since CCR will simply ignore DLS/FLS so the
```

Comment:
To get CCR coverage, we could either copy/paste from RemoteClusterSecurityCcrIT or possible just include another test case there